crates/nu-cli/tests/commands/sum.rs

```diff
@@ -60,3 +60,162 @@ fn outputs_zero_with_no_input() {
+    Playground::setup("sum_test_3", |dirs, sandbox| {
+        sandbox.with_files(vec![FileWithContentToBeTrimmed(
+            "calendar.json",
+            r#"
+                    [
+                        {
+                            "sunday": null,
+                            "monday": null,
+                            "tuesday": null,
+                            "wednesday": null,
+                            "thursday": null,
+                            "friday": 1,
+                            "saturday": 2
+                        },
+                        {
+                            "sunday": 3,
+                            "monday": 4,
+                            "tuesday": 5,
+                            "wednesday": 6,
+                            "thursday": 7,
+                            "friday": 8,
+                            "saturday": 9
+                        },
+                        {
+                            "sunday": 10,
+                            "monday": 11,
+                            "tuesday": 12,
+                            "wednesday": 13,
+                            "thursday": 14,
+                            "friday": 15,
+                            "saturday": 16
+                        },
+                        {
+                            "sunday": 17,
+                            "monday": 18,
+                            "tuesday": 19,
+                            "wednesday": 20,
+                            "thursday": 21,
+                            "friday": 22,
+                            "saturday": 23
+                        },
+                        {
+                            "sunday": 24,
+                            "monday": 25,
+                            "tuesday": 26,
+                            "wednesday": 27,
+                            "thursday": 28,
+                            "friday": 29,
+                            "saturday": 30
+                        },
+                        {
+                            "sunday": 31,
+                            "monday": null,
+                            "tuesday": null,
+                            "wednesday": null,
+                            "thursday": null,
+                            "friday": null,
+                            "saturday": null
+                        }
+                    ]
+                "#,
+        )]);
+        let answers_for_rows = ["3", "42", "91", "140", "189", "0"];
+        for (row_idx, answer) in answers_for_rows.iter().enumerate() {
+            let actual = nu!(
+                cwd: dirs.test(), pipeline(
+                    &format!(r#"
+                        open "calendar.json"
+                        | nth {}
+                        | sum
+                    "#, row_idx)
+            ));
+            assert_eq!(actual.out, *answer);
+        }
+    })
+}
+
+#[test]
+fn compute_sum_of_multiple_rows() {
```

Comment:
Unless my eyes are deceiving me, this test looks exactly like the one above?

Reply:
The difference is here:
```rust
 let actual = nu!(
                cwd: dirs.test(), pipeline(
                    &format!(r#"
                        open "calendar.json"
                        | nth {}
                        | sum
                    "#, row_idx)
            ));
```
versus
```rust
let actual = nu!(
                cwd: dirs.test(), pipeline(
                    &format!(r#"
                        open "calendar.json"
                        | sum
                        | nth {}
                    "#, row_idx)
            ));
```
Note that in the first example the output of `nth` is piped to sum while in the second the output of `sum` is piped to nth. Effectively the first is testing that sum correctly computes the output of a single row, while the second is testing the sum of a table is itself a row containing the sum of each of the rows in the table. 

I can factor out commonalities from the two functions or add some comments into the source to make the distinction clear.

Reply:
Ahhh, essentially showing a kind of "commutative" property of `sum`. No need to refactor 🙂